docs/proposals/acting-as.md

```diff
@@ -0,0 +1,219 @@
+- quota controller: not sure if needed to use "Acting-As".
+- namespace controller: if it is executing initializers and finalizers, then it may need to "Acting-As" for some of those steps.  Needs thought.
+- horizontal-scaler controller: should use "Acting-As" to ensure it does not scale things the HPA creator should not be able to scale.
+- persistent-volume controller: not sure if needed to use "Acting-As".
+- persistent-volume recycler: not sure if needed to use "Acting-As"
+- serviceaccount.TokensController: probably should, since it CRUDs objects in response to changed made in other objects by users.
+- serviceaccount.ServiceAccountController: probably should, since it CRUDs objects in response to changed made in other objects by users.
+
+These do not all need to be convered in one go.  It should be possible to have an incremental transition.
+
+## Alternatives considered
+
+### Alternative: Access Review API
+
+In this alternate design, Controllers would not use Acting-As header, and they would not have
+authorization checks compiled into them.  Instead, they use an "Access Review" interface on the
+APIserver that lets a controller ask "If I were user A would I be allowed to create this Pod?".
+
+Problem with this is that a system admin has to fully trust any new controller (e.g. contributed
+controller).
+
+Recommendation: reject this alternative.
+
+### Alternative: Special-purpose accounts
+
+In this alternate design, the controller-manager authenticates
+as a special-purpose account called "controller-manager-acting-in-namespace-foo".
+No "acting-as" header.  Controller-manager actually stores credentials for all
+the accounts it is allowed to act as, and when a namespace is created,
+either the namespace admin consents to let controller-manager act in it,
+or a namespace initializer process creates the account.
+
+This is a good approach too.
+It is analgous to creating a "bot" account on our github organization, and then adding that
+bot account to the ACLs of our github repo.
+
+Good things about this approach:
+
+1. no special purpose header needed
+1. no need to record two users in an audit log (original and acting-as).
+1. no need for second authorization stage in the apiserver.
+1. since the special-purpose account (e.g. "controller-manager-acting-in-namespace-foo")
+   is added to the ACL for a particular namespace, a Policy Manager for that namespace
+   could revoke that ACL if desired.
+
+Concerns about this approach:
+
+1. more complex to get right at first.
+1. requires that users see a special purpose account added to the ACL of their namespace/cluster
+   for each use-case.  
+1. workflow for automating adding these to new namespaces could get complex.  Even more
+   so when there are multiple roles in a namespace.
+1. automation of the workflow for adding these special-purpose accounts to ACLs requires
+   a standardized ACL API for kubernetes, which competes with the goal of pluggable
+   authorization in kubernetes.
```

Comment:
If openshift authorization is the sole authorization method, then yes.  As a hack, you could run a union of authorizers which always says yes to authorizing system users to act as default service accounts.